src/check/property/SkipEqualValuesWithComparatorProperty.ts

```diff
@@ -0,0 +1,19 @@
+import { IRawProperty } from './IRawProperty';
+import { Random } from '../../random/generator/Random';
+import { Shrinkable } from '../arbitrary/definition/Shrinkable';
+
+export class SkipEqualValuesWithComparatorProperty<Ts, IsAsync extends boolean> implements IRawProperty<Ts, IsAsync> {
+  private coveredValues: Ts[] = [];
+
+  constructor(readonly property: IRawProperty<Ts, IsAsync>, readonly comparator: (a: Ts, b: Ts) => boolean) {}
+
+  isAsync = (): IsAsync => this.property.isAsync();
+  generate = (mrng: Random, runId?: number): Shrinkable<Ts> => this.property.generate(mrng, runId);
+  run = (v: Ts): ReturnType<IRawProperty<Ts, IsAsync>['run']> => {
+    if (this.coveredValues.some((coveredValue) => this.comparator(coveredValue, v))) {
+      return null as any;
```

Comment:
Return success status instead (no `as` required if return type is OK)

Reply:
![image](https://user-images.githubusercontent.com/7964583/110220740-bafb5000-7ed8-11eb-86e6-7e1530675a8e.png)
I didn't see how to return success status in other way than null.

Reply:
I thought back about it. Actually, the current PR not really "skip" the run (with fast-check meaning) but rather do not run the code of the property. As it does not skip it has to return a status: either success or failure, and actually it can be both of them. I'd say that the best option will be to return the returned value associated to this stringified value. We could for instance have something like:

```js
// alreadySeenValuesAndReturn: Map<string, ReturnType<IRawProperty...[run]>

const stringifiedValue = stringified(v)
if (this.alreadySeenValuesAndReturn.has(stringifiedValue))
  return this.alreadySeenValuesAndReturn.get(stringifiedValue)!;

const out = this.property.run(v);
this.alreadySeenValuesAndReturn.set(stringifiedValue, out)
return out;
```

Why that? Because with this code we could bypass a failed run (it may happen to replay some already tested values during shrinking process, some of those replays may be related to red runs, in that case in order to keep a clear error report, the run should not be marked as success and return the original version).

My snippet above is not perfect (mostly for sync property throwing stuff, the code should rethrow it to be consistent*). *the feature is like 'do not re-run user code but behave as if it was executed' 

_from my mobile_